File: string/student-attendance-record-i.cpp
class Solution {
public:
    bool checkRecord(string s) {
        // init
        int absent = 0;
        int late = 0;

        // check each day
        for (char c : s) {
            if (c == 'A') {
                absent++;
                late = 0;
            }
            else if (c == 'L') {
                late++;
            }
            else {
                late = 0;
            }

            // check eligibility
            if (absent >= 2 || late >= 3) {
                return false;
            }
        }

        return true;
    }
    }
};

File: string/student-attendance-record-i.py
class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        # init
        absent = 0
        late = 0

        # check each day
        for c in s:
            if c == 'A':
                absent += 1
                late = 0

            elif c == 'L':
                late += 1

            else:
                late = 0

            # check eligibility
            if absent >= 2 or late >= 3:
                return False

        return True